cmd/node-cache/main.go

```diff
@@ -163,7 +163,7 @@ func (c *cacheApp) parseAndValidateFlags() error {
 	flag.StringVar(&c.params.localIP, "localip", "", "ip address to bind dnscache to")
 	flag.StringVar(&c.params.interfaceName, "interfacename", "nodelocaldns", "name of the interface to be created")
 	flag.DurationVar(&c.params.interval, "syncinterval", 60, "interval(in seconds) to check for iptables rules")
-	flag.IntVar(&c.params.metricsPort, "metricsport", 9353, "port to serve nodecache setup metrics")
+	flag.StringVar(&c.params.metricsListenAddress, "metrics-listen-address", "127.0.0.2:9353", "address to serve metrics on")
```

Comment:
Was this value verified to work? Maybe we can make it listen on 0.0.0.0: by default?

Reply:
Yes I did test it, it does work. The reason I didn't choose `0.0.0.0` by default is that there is a chance something is already listening on port 9353 which would make the cash crash.

LMK what you think.

Reply:
If some other service binds to 0.0.0.0:9353 first, this would still fail correct?

Instead, suppose a different service bound to <node-ip>:9353. And if this also exposed metrics on /metrics. What would accessing "http://node-ip:9353/metrics" give us? The metrics of node-local-dns or this other service?

Did you verify by using the node-ip to scrape the metrics?

Reply:
> If some other service binds to 0.0.0.0:9353 first, this would still fail correct?

Yes, good point

> Instead, suppose a different service bound to :9353. And if this also exposed metrics on /metrics. What would accessing "http://node-ip:9353/metrics" give us? The metrics of node-local-dns or this other service?

The other service, obviously